Mage/src/main/java/mage/target/TargetImpl.java

```diff
@@ -595,4 +601,31 @@ public int getSize() {
     public boolean contains(UUID targetId) {
         return targets.containsKey(targetId);
     }
+
+    @Override
+    public UUID tryToAutoChoose(UUID abilityControllerId, Ability source, Game game) {
+        Set<UUID> possibleTargets = possibleTargets(abilityControllerId, source, game);
+        possibleTargets.removeAll(this.targets.keySet());
+        return tryToAutoChoose(abilityControllerId, source, game, possibleTargets);
+    }
+
+    @Override
+    public UUID tryToAutoChoose(UUID abilityControllerId, Ability source, Game game, Collection<UUID> possibleTargets) {
+        Player player = game.getPlayer(abilityControllerId);
+        boolean strictModeEnabled = player != null && player.getStrictChooseMode();
```

Comment:
@JayDi85 I have added an override for the test player.

The test with `setChooseStrictMode` will fail since I haven't change. Should I:
1. Disable `setChooseStrictMode` for the tests.
2. Undo the changes I did to make them work.

Reply:
I am undoing most, except ones like the SimpleDominariaCards which required a try-catch block otherwise since that is hacky.